homeassistant/components/fyta/config_flow.py

```diff
@@ -30,36 +32,80 @@ class FytaConfigFlow(ConfigFlow, domain=DOMAIN):
+        """Reusable Auth Helper."""
+        credentials: dict[str, str | datetime] = {}
+        fyta = FytaConnector(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+
+        try:
+            credentials = await fyta.login()
+        except FytaConnectionError:
+            return {"base": "cannot_connect"}
+        except FytaAuthentificationError:
+            return {"base": "invalid_auth"}
+        except FytaPasswordError:
+            return {"base": "invalid_auth", CONF_PASSWORD: "password_error"}
+        except Exception as e:  # pylint: disable=broad-except
+            _LOGGER.error(e)
+            return {"base": "unknown"}
+        finally:
+            await fyta.client.close()
+
+        if isinstance(credentials["expiration"], datetime):
+            credentials["expiration"] = credentials["expiration"].isoformat()
+        self.credentials = credentials
+
+        return {}
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Handle the initial step."""
-
         errors = {}
         if user_input:
             self._async_abort_entries_match({CONF_USERNAME: user_input[CONF_USERNAME]})
 
-            fyta = FytaConnector(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
-
-            try:
-                await fyta.login()
-            except FytaConnectionError:
-                errors["base"] = "cannot_connect"
-            except FytaAuthentificationError:
-                errors["base"] = "invalid_auth"
-            except FytaPasswordError:
-                errors["base"] = "invalid_auth"
-                errors[CONF_PASSWORD] = "password_error"
-            except Exception:  # pylint: disable=broad-except
-                errors["base"] = "unknown"
-            else:
+            if not (errors := await self.async_auth(user_input)):
+                user_input |= self.credentials
                 return self.async_create_entry(
                     title=user_input[CONF_USERNAME], data=user_input
                 )
-            finally:
-                await fyta.client.close()
 
         return self.async_show_form(
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Handle flow upon an API authentication error."""
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reauthorization flow."""
+        errors = {}
+        assert self._entry is not None
+
+        if user_input and not (errors := await self.async_auth(user_input)):
+            user_input |= self.credentials
```

Comment:
This isn't needed in the user step so why we need to merge in credentials in reconfigure?

Reply:
I also added it for the user step (line 72).

It is not yet read in `async_setup_entry`, as I understood this should be added in a separate PR.

Reply:
Not sure I understand. If it wasn't needed before why it's needed now?

Reply:
Still in `async_auth`

Reply:
Sorry, missed this.